common/djangoapps/third_party_auth/api/utils.py

```diff
@@ -0,0 +1,29 @@
+"""
+Shareable utilities for third party auth api functions
+"""
+
+from social_django.models import UserSocialAuth
+
+
+def filter_user_social_auth_queryset_by_provider(query_set, provider):
+    """
+    Filter a query set by the given TPA provider
+
+    Params:
+        query_set: QuerySet[UserSocialAuth]
+        provider: common.djangoapps.third_party_auth.models.ProviderConfig
+    Returns:
+        QuerySet[UserSocialAuth]
+    """
+    query_set = query_set.filter(provider=provider.backend_name)
+
+
+    # build our query filters
+    # When using multi-IdP backend, we only retrieve the ones that are for current IdP.
+    # test if the current provider has a slug
+    uid = provider.get_social_auth_uid('uid')
```

Comment:
These 4 lines doesn't seem to belong in this function. Can you please help explain how they are relevant to the util function?

Reply:
Also, in the comments, can you be straightforward and say the purpose of these test is to identify SAMLProvider vs other type of providers (oauth2), so you can provide the correct UID filter?

Reply:
Does this help?

Reply:
Yes. The comments update did help!